teuthology/task/install/rpm.py

```diff
@@ -62,6 +64,45 @@ def _remove(ctx, config, remote, rpm):
     else:
         remote.run(args='sudo yum clean expire-cache')
 
+def _yum_addrepo(remote, repo_list):
+    """
+    Add yum repos to the remote system.
+
+    :param remote: remote node where to add packages
+    :param repo_list: list of dictionaries with keys 'name', 'url', etc.
+    :return:
+    """
+    for repo in repo_list:
+        filepath = "/etc/yum.repo.d/{}.repo".format(repo['name'])
+        args=['sudo', 'touch', filepath]
+        remote.run(args=args)
+        args=['sudo', 'echo', '[{}]'.format(repo['name']), run.Raw('>'), filepath]
+        remote.run(args=args)
+        args=['sudo', 'echo', 'name={}'.format(repo['name']), run.Raw('>>'), filepath]
+        remote.run(args=args)
+        args=['sudo', 'echo', 'enabled=1', run.Raw('>>'), filepath]
+        remote.run(args=args)
+        args=['sudo', 'echo', 'baseurl={}'.format(repo['url']), run.Raw('>>'), filepath]
+        remote.run(args=args)
```

Comment:
You have python but using this shell on the remote node?
I would suggest to compose repo data and just write to a remote file, for example:
```
    yum_repo_text = (
         f"[{repo['name']}]\n"
         f"name={repo['name']}\n"
         f"enabled=1\n"
         f"baseurl={repo['url']}\n")
    if 'gpgcheck' in repo:
         yum_repo_text += (
              f"gpgcheck={repo['gpgcheck']}\n"
              f"gpgkey={repo['gpgkey']}\n")
    remote.write_file(yum_repo_file, yum_repo_text, sudo=True)
```

Reply:
but as I said earlier, since we are using `zypper addrepo` command for zypper, can we use similar `yum` command for repos manipulation?

Reply:
agreed. `dnf config-manager --add-repo` and `dnf config-manager --setopt` should do the trick.

Reply:
Yeah, sure. Will switch to `dnf config-manager --add-repo`.